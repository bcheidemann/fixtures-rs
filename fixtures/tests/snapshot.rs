use indoc::indoc;
use std::process::Command;

fn test_example_expansion(example_name: &str) {
    let output = Command::new("cargo")
        .args(["expand", "--tests", "--example", example_name])
        .output()
        .expect("failed to expand example");

    if !output.status.success() {
        let stdout = String::from_utf8_lossy(&output.stdout);
        let stderr = String::from_utf8_lossy(&output.stderr);
        panic!(
            indoc! {"
                error: failed to expand example ({})

                -- begin stdout --
                {}
                -- end stdout --

                -- begin stderr --
                {}
                -- end stderr --
            "},
            example_name, stdout, stderr
        );
    }

    let current_dir = std::env::current_dir()
        .expect("failed to get current directory")
        .to_string_lossy()
        .to_string();
    let expansion = String::from_utf8_lossy(&output.stdout).replace(&current_dir, "<repo>");

    insta::assert_snapshot!(example_name, expansion);
}

#[test]
fn basic_usage() {
    test_example_expansion("basic_usage");
}

#[test]
fn complex_signatures() {
    test_example_expansion("complex_signatures");
}

#[test]
fn criterion() {
    test_example_expansion("criterion");
}

#[test]
fn ignore_attributes() {
    test_example_expansion("ignore_attributes");
}

#[test]
fn multiple_fixtures() {
    test_example_expansion("multiple_fixtures");
}

#[test]
fn negative_globs() {
    test_example_expansion("negative_globs");
}
