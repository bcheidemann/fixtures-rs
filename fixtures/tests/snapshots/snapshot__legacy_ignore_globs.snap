---
source: fixtures/tests/snapshot.rs
expression: expansion
---
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use std::path::Path;
use fixtures::fixtures;
fn test1(_path: &Path) {}
mod test1 {
    use super::*;
    extern crate test;
    #[rustc_test_marker = "test1::file_1_dot_txt"]
    #[doc(hidden)]
    pub const file_1_dot_txt: test::TestDescAndFn = test::TestDescAndFn {
        desc: test::TestDesc {
            name: test::StaticTestName("test1::file_1_dot_txt"),
            ignore: false,
            ignore_message: ::core::option::Option::None,
            source_file: "fixtures/examples/legacy_ignore_globs.rs",
            start_line: 14usize,
            start_col: 4usize,
            end_line: 14usize,
            end_col: 9usize,
            compile_fail: false,
            no_run: false,
            should_panic: test::ShouldPanic::No,
            test_type: test::TestType::Unknown,
        },
        testfn: test::StaticTestFn(
            #[coverage(off)]
            || test::assert_test_result(file_1_dot_txt()),
        ),
    };
    pub fn file_1_dot_txt() {
        test1(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/legacy_ignore_globs/file_1.txt",
            ),
        )
    }
    extern crate test;
    #[rustc_test_marker = "test1::file_2_dot_ignore_dot_txt"]
    #[doc(hidden)]
    pub const file_2_dot_ignore_dot_txt: test::TestDescAndFn = test::TestDescAndFn {
        desc: test::TestDesc {
            name: test::StaticTestName("test1::file_2_dot_ignore_dot_txt"),
            ignore: true,
            ignore_message: ::core::option::Option::Some("reason for ignoring file"),
            source_file: "fixtures/examples/legacy_ignore_globs.rs",
            start_line: 14usize,
            start_col: 4usize,
            end_line: 14usize,
            end_col: 9usize,
            compile_fail: false,
            no_run: false,
            should_panic: test::ShouldPanic::No,
            test_type: test::TestType::Unknown,
        },
        testfn: test::StaticTestFn(
            #[coverage(off)]
            || test::assert_test_result(file_2_dot_ignore_dot_txt()),
        ),
    };
    #[ignore = "reason for ignoring file"]
    pub fn file_2_dot_ignore_dot_txt() {
        test1(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/legacy_ignore_globs/file_2.ignore.txt",
            ),
        )
    }
    pub const EXPANSIONS: &[fn()] = &[file_1_dot_txt, file_2_dot_ignore_dot_txt];
}
fn test2(_path: &Path) {}
mod test2 {
    use super::*;
    extern crate test;
    #[rustc_test_marker = "test2::file_1_dot_txt"]
    #[doc(hidden)]
    pub const file_1_dot_txt: test::TestDescAndFn = test::TestDescAndFn {
        desc: test::TestDesc {
            name: test::StaticTestName("test2::file_1_dot_txt"),
            ignore: false,
            ignore_message: ::core::option::Option::None,
            source_file: "fixtures/examples/legacy_ignore_globs.rs",
            start_line: 21usize,
            start_col: 4usize,
            end_line: 21usize,
            end_col: 9usize,
            compile_fail: false,
            no_run: false,
            should_panic: test::ShouldPanic::No,
            test_type: test::TestType::Unknown,
        },
        testfn: test::StaticTestFn(
            #[coverage(off)]
            || test::assert_test_result(file_1_dot_txt()),
        ),
    };
    pub fn file_1_dot_txt() {
        test2(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/legacy_ignore_globs/file_1.txt",
            ),
        )
    }
    extern crate test;
    #[rustc_test_marker = "test2::file_2_dot_ignore_dot_txt"]
    #[doc(hidden)]
    pub const file_2_dot_ignore_dot_txt: test::TestDescAndFn = test::TestDescAndFn {
        desc: test::TestDesc {
            name: test::StaticTestName("test2::file_2_dot_ignore_dot_txt"),
            ignore: true,
            ignore_message: ::core::option::Option::None,
            source_file: "fixtures/examples/legacy_ignore_globs.rs",
            start_line: 21usize,
            start_col: 4usize,
            end_line: 21usize,
            end_col: 9usize,
            compile_fail: false,
            no_run: false,
            should_panic: test::ShouldPanic::No,
            test_type: test::TestType::Unknown,
        },
        testfn: test::StaticTestFn(
            #[coverage(off)]
            || test::assert_test_result(file_2_dot_ignore_dot_txt()),
        ),
    };
    #[ignore]
    pub fn file_2_dot_ignore_dot_txt() {
        test2(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/legacy_ignore_globs/file_2.ignore.txt",
            ),
        )
    }
    pub const EXPANSIONS: &[fn()] = &[file_1_dot_txt, file_2_dot_ignore_dot_txt];
}
#[allow(dead_code)]
fn main() {}
#[rustc_main]
#[coverage(off)]
#[doc(hidden)]
pub fn main() -> () {
    extern crate test;
    test::test_main_static(
        &[
            &file_1_dot_txt,
            &file_2_dot_ignore_dot_txt,
            &file_1_dot_txt,
            &file_2_dot_ignore_dot_txt,
        ],
    )
}
